### PR TITLE
prov/verbs, rxm: bug fixes + minor refactoring

### PR DESCRIPTION
**Pull request by a-ilango**

### CODE REVIEW

File: src/common.c [hunk 1]
```diff
@@ -555,6 +555,35 @@ int ofi_addr_cmp(const struct fi_provider *prov, const struct sockaddr *sa1,
 	}
 }
 
+int ofi_is_only_port_set(const char *node, const char *service,
```
Comment (shefty) [inline src/common.c]: I would rename to only_src_port_set, since the function returns 0 if a dest_addr is provided

File: include/ofi_net.h [hunk 2]
```diff
@@ -226,6 +226,38 @@ static inline int ofi_is_any_addr(struct sockaddr *sa)
 	}
 }
 
+static inline int ofi_addr_get_port(void *addr)
```
Comment (shefty) [inline include/ofi_net.h]: Change return value to uint16_t

**Comment by shefty** [comment]
See PR #3720 -- which contains the same fix, but with an additional change.

**Comment by a-ilango** [comment]
> See PR #3720 -- which contains the same fix, but with an additional change.

Thanks. I guess I'll drop the same commit in this PR since the other one has an additional change.

Btw, did you add a merge commit to this PR?

**Comment by shefty** [comment]
Something changed recently that I'm guessing is related to making DCO required.  As soon as any PR is merged, then all open PRs can no longer be merged without forcing an update on the branch.  So I clicked on the 'Update branch' button to make this PR merge-able, which looks like adds a merge commit to the PR.

Let me talk with @jsquyres about this side effect of making DCO required.  I think this only changes the PR and not anyone's branch (at least I hope so!)

File: src/common.c (hunk 1 above)
Comment (a-ilango) [reply src/common.c]: Sure. Also I should add a check for flags since node and service would indicate source only when FI_SOURCE is set.

**Comment by a-ilango** [comment]
I've updated the PR with requested changes.

**Pull request merged**